### ISSUE TITLE
The training result is blurry, please help

### ISSUE THREAD

**Issue by meeio** [junyanz/pytorch-CycleGAN-and-pix2pix#226]
Hi, I really like your project, so I implement it with Tensorflow.

But when I train on horse2zebra dataset it not went well.

I use ngf=64 with 9 residual block, udf=64 with 4 layers, when I train the model at first 90k steps, the generated image is pretty clear, and some of it has great result, than I want train more to get better result, but when the training steps come up to 110k, the result is become blurry, just like this:
![test](https://user-images.githubusercontent.com/12381290/37871831-43809ada-302a-11e8-9f1f-9f40c76ea638.PNG)

Is that because I did not use identity mapping or something else?

It seems to me that the Generator cannot remember the structure of original image, what if adding some skip connection to the Generator network will mitigate the issue?

**Comment by junyanz** [comment]
It's hard to tell without reading your codebase. Harry provided a good Tensorflow [implementation](https://github.com/leehomyc/cyclegan-1). Maybe you want to compare the difference between his and yours.

**Comment by meeio** [comment]
thanks for your reply, I can reach good result in my project as well
![14](https://user-images.githubusercontent.com/12381290/38010082-fc799904-3289-11e8-9e9c-d5d8017c8818.PNG)
The result is training after 83000 steps, and it pretty good I think, but when it reach to 130000 steps, the result is become  blurry and twisty, is that normal or just in my case?

**Comment by meeio** [comment]
@junyanz Is that because of overfitting? The generator network overfit the distribution of target domain, so that  the result become blurry? thanks!

**Comment by junyanz** [comment]
Did you reduce the learning rate in the later stage of the training?

**Comment by zoucheng1991** [comment]
@junyanz  Hi, I'm also getting this issue, using ngf=64 with 7 residual block, udf=64 with 2 layers,  I keep lr for the initial 20 epochs and decay lr to 0 in the following 20 epochs. After about 25 epochs, the result seems good, but after I trained all the 40 epochs, the result is a little blurred.
Could this be caused by overfitting? And can dropout help it?

**Comment by junyanz** [comment]
Would you be able to get good results with the original parameters in the paper? (loadSize=286, fineSize=256, netG=resnet_9blocks, netD=default)

**Comment by zoucheng1991** [comment]
> Would you be able to get good results with the original parameters in the paper? (loadSize=286, fineSize=256, netG=resnet_9blocks, netD=default)

As I'm working on loadSize and fineSize 640，due to limiation of GPU mem of 1080ti, I can use no more than 7 residual block😂

**Comment by junyanz** [comment]
See  [Traing/test tips](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/tips.md#trainingtesting-with-high-res-images)  and this [Q & A](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/qa.md#out-of-memory-174) for how to reduce memory when training with high-res images.

**Comment by zoucheng1991** [comment]
@junyanz thx a lot, that should be of great help
